### PR TITLE
ZKUI-330: Storage Consumption - useListAccount hook test 

### PR DESCRIPTION
**Pull request by ChengYanJin**

### CODE REVIEW

File: src/react/next-architecture/adapters/metrics/MockedMetricsAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  listLocationsLatestUsedCapacity(
+    locationIds: string[],
+  ): Promise<LocationLatestUsedCapacity[]> {
+    throw new Error('Method not implemented.');
+  }
+  listAccountLocationsLatestUsedCapacity(
+    accountCanonicalId: string,
+  ): Promise<LocationLatestUsedCapacity[]> {
+    throw new Error('Method not implemented.');
+  }
```
Comment (JBWatenbergScality) [inline src/react/next-architecture/adapters/metrics/MockedMetricsAdapter.ts:30]: in the end each of those methods should be jest.fn but it's ok for now

File: src/react/next-architecture/domain/business/accounts.test.tsx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+  it('should return accounts as soon as it is resolved', () => {
+    //S
+    const accountsAdapter = new MockedAccountsAdapter();
+    const metricsAdapter = new MockedMetricsAdapter();
+    metricsAdapter.listAccountsLatestUsedCapacity = jest
+      .fn()
+      .mockImplementation(async () => {
+        setTimeout(() => {
+          return [];
+        }, 5000);
```
Comment (JBWatenbergScality) [inline src/react/next-architecture/domain/business/accounts.test.tsx:29]:
```suggestion
        return new Promise((resolve) => setTimeout(() => resolve([]), 5000);
```

File: src/react/next-architecture/domain/business/accounts.test.tsx [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        useAccountLatestUsedCapacity({
+          metricsAdapter,
+          accountCanonicalId: ACCOUNT_CANONICAL_ID,
+        }),
+      {
+        wrapper,
+      },
+    );
+    //V
+    expect(result.current.usedCapacity.status).toBe('idle');
```
Comment (JBWatenbergScality) [inline src/react/next-architecture/domain/business/accounts.test.tsx:261]:
```suggestion
    expect(result.current.usedCapacity.status).toBe('success');
```

File: src/react/next-architecture/domain/business/accounts.test.tsx [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    );
+    //V
+    await waitForMetrics(() => result.current.accounts.status === 'success');
+
+    expect(
+      //@ts-ignore
+      result.current.accounts.value(
+        (account: Account) => account.canonicalId === ACCOUNT_CANONICAL_ID,
+      ).usedCapacity.status,
+    ).toBe('idle');
```
Comment (JBWatenbergScality) [inline src/react/next-architecture/domain/business/accounts.test.tsx:292]:
```suggestion
    ).toBe('success');
```

**Review by JBWatenbergScality — APPROVED** [review]
LGTM

File: src/react/next-architecture/domain/business/accounts.test.tsx (hunk 3 above)
Comment (ChengYanJin) [reply src/react/next-architecture/domain/business/accounts.test.tsx:261]: Yes. I'll also check that the status won't go through `loading`.

**Pull request merged**